policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +68,52 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for container in spec.iter().flat_map(|s| s.containers.iter()) {
+        set_probe_ports(&container.liveness_probe, &mut probes, port_names);
+        set_probe_ports(&container.readiness_probe, &mut probes, port_names);
+        set_probe_ports(&container.startup_probe, &mut probes, port_names);
+    }
+    probes
+}
+
+fn set_probe_ports(
+    probe: &Option<k8s::Probe>,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    let probe = match probe {
+        Some(probe) => probe,
+        None => return,
+    };
+    if let Some(http) = &probe.http_get {
+        if let Some(path) = &http.path {
```

Comment:
What does it mean if an HTTP probe doesn't have a path set, exactly? It seems like this would imply `/`?

Reply:
That's correct it gets automatically set to `/` which means that if there is an `httpGet` then there should be a `path` field. I tested this by creating a Pod with an empty `httpGet.path`, and when it's created it's set to `/`.

I could have changed this to `unwrap()` since `path` should exist, but I decided to just go with the `if let`. I can change it to `unwrap()` if you want though.

Reply:
Changed to `expect()` in [284f1ab](https://github.com/linkerd/linkerd2/pull/8961/commits/284f1ab8c00371788ed300e9c7e63132aa6c96a7)

Reply:
I don't think panicking is appropriate. If the API says the field is optional, we should handle the case when the value is missing without panicking.